src/main/java/org/wildfly/security/http/impl/DigestAuthenticationMechanism.java

```diff
@@ -155,7 +158,19 @@ private void validateResponse(HashMap<String, byte[]> responseTokens, final Http
         byte[] salt = messageRealm.getBytes(UTF_8);
         boolean nonceValid = nonceManager.useNonce(nonce, salt, nonceCount);
 
-        String username = convertToken(USERNAME, responseTokens.get(USERNAME));
+        String username;
```

Comment:
Just idea. As this star thing could be in theory applied to any attribute. Could this be refactored to util method with attribute name parameter